src/components/store/AddCartButton/index.tsx

```diff
@@ -0,0 +1,91 @@
+import { Dropdown } from '@/components/common';
+import { useEffect, useId, useState } from 'react';
+import styles from './style.module.scss';
+
+interface AddCartButtonProps {
+  currSize: string | undefined;
+  // Justification for disabling rules: This seems to be a false positive.
+  // https://stackoverflow.com/q/63767199/
+  // eslint-disable-next-line no-unused-vars
+  inStock: boolean;
+  inCart: boolean;
+  lifetimeRemaining: number;
+  montlyRemaining: number;
+  setInCart: (value: boolean) => void;
```

Comment:
in React's naming conventions, starting something with `set` should be reserved for something from `useState` within a component. if you want to pass the new `inCart` value back up to the parent component, the prop should start with `on`, like an event handler

```suggestion
  onCartChange: (inCart: boolean) => void;
```

Reply:
<img width="332" alt="image" src="https://github.com/acmucsd/membership-portal-ui-v2/assets/54158686/8c2c0489-8e21-4ae9-a69e-9c919647684d">